edge/forwarding.go

```diff
@@ -0,0 +1,103 @@
+package edge
+
+// ForwardReceiver handles messages as they arrive and can return a message to be forwarded to output edges.
+// If a returned messages is nil, no message is forwarded.
+type ForwardReceiver interface {
```

Comment:
Why don't all of the nodes use a `ForwardReceiver`? Reading through the [combine.go](https://github.com/influxdata/kapacitor/blob/nc-mpi/combine.go), it seems like it would have been possible to implement it using a `ForwardReceiver` instead of a `Receiver`. Am I missing something?